### DISCUSSION TITLE
聚类训练报错ValueError: need at least one array to concatenate

### DISCUSSION THREAD

**Discussion by yszzQ** [svc-develop-team/so-vits-svc#297] [Q&A]
### 请勾选下方的确认框。

- [X] 我已仔细阅读[README.md](https://github.com/svc-develop-team/so-vits-svc/blob/4.0/README_zh_CN.md)和[wiki中的Quick solution](https://github.com/svc-develop-team/so-vits-svc/wiki/Quick-solution)。
- [X] 我已通过各种搜索引擎排查问题，我要提出的问题并不常见。
- [X] 我未在使用由第三方用户提供的一键包/环境包。

### 系统平台版本号

win10

### GPU 型号

3090ti

### Python版本

3.9

### PyTorch版本

11.7

### sovits分支

4.0(默认)

### 数据集来源（用于判断数据集质量）

自用

### 出现问题的环节或执行的命令

python cluster/train_cluster.py

### 问题描述

报错

### 日志

```python
train kmeans for .ipynb_checkpoints...
INFO:__main__:Loading features from dataset/44k/.ipynb_checkpoints
0it [00:00, ?it/s]
Traceback (most recent call last):
  File "cluster/train_cluster.py", line 64, in <module>
    x = train_cluster(in_dir, n_clusters, verbose=False)
  File "cluster/train_cluster.py", line 24, in train_cluster
    features = np.concatenate(features, axis=0)
  File "<__array_function__ internals>", line 180, in concatenate
ValueError: need at least one array to concatenate
```

### 截图`so-vits-svc`、`logs/44k`文件夹并粘贴到此处

/

**Comment by PriesiaMioShirakana** [comment]
你有没有先预处理数据集，数据集要先预处理才可以训练cluster

**Comment by PriesiaMioShirakana** [comment]
新联cluster时会直接读取soft.pt文件，而这个soft.pt文件是预处理时生成的

**Comment by yszzQ** [comment]
> 你有没有先预处理数据集，数据集要先预处理才可以训练cluster

我是模型练完接着就开始这一步了，还要重新预处理数据集吗？

**Comment by PriesiaMioShirakana** [comment]
查看一下你的dataset/44k/.ipynb_checkpoints文件夹是否有文件

**Comment by yszzQ** [comment]
> 

谢谢 解决了

**Comment by HuanLinOTO** [comment]
it's look lite a problem of using, concert to discussion
